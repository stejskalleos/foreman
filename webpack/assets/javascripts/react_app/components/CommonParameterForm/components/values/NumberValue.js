import React from 'react';
import PropTypes from 'prop-types';

const NumberValue = ({ id, name, css, value, onChange, isMasked }) => (
  <input
    type="number"
    id={id}
    name={name}
    className={`${css} ${isMasked && 'masked-input'}`}
    value={value}
    onChange={e => onChange(e.target.value)}
  />
);

NumberValue.propTypes = {
  id: PropTypes.string,
  name: PropTypes.string.isRequired,
  css: PropTypes.string,
  value: PropTypes.any,
  onChange: PropTypes.func.isRequired,
  isMasked: PropTypes.bool,
};

NumberValue.defaultProps = {
  id: '',
  css: '',
  value: '',
  isMasked: false,
};

export default NumberValue;
