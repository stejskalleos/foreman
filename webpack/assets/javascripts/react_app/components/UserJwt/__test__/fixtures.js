import { noop } from '../../../common/helpers';
import { STATUS } from '../../../constants';

export const formProps = {
  handleSubmit: noop,
};

const token = 'token23';

export const fieldOk = {
  apiStatus: STATUS.RESOLVED,
  token,
};

export const fieldError = {
  apiStatus: STATUS.ERROR,
  token,
};

export const fieldPending = {
  apiStatus: STATUS.PENDING,
  token,
};
