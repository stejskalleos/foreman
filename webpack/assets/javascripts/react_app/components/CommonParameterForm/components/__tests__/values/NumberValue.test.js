import { testComponentSnapshotsWithFixtures } from '@theforeman/test';

import { numberValueFixtures } from '../fixtures';
import NumberValue from '../../values/NumberValue';


describe('NumberValue', () => {
  describe('rendering', () =>
    testComponentSnapshotsWithFixtures(NumberValue, numberValueFixtures));
});
