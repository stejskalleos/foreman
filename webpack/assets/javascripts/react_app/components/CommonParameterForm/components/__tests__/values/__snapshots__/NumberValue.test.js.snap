// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NumberValue rendering default 1`] = `
<input
  className="form-control false"
  id="common_parameter_value"
  name="common_parameter[value]"
  onChange={[Function]}
  type="number"
  value="123"
/>
`;
